clients/da-vinci-client/src/test/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTaskTest.java

```diff
@@ -4028,6 +4033,111 @@ public void testMaybeSendIngestionHeartbeat(
+    ReadOnlyStoreRepository mockReadOnlyStoreRepository = mock(ReadOnlyStoreRepository.class);
+    doReturn(mockStore).when(mockReadOnlyStoreRepository).getStoreOrThrow(eq(storeName));
+    doReturn(false).when(mockStore).isHybridStoreDiskQuotaEnabled();
+    doReturn(Optional.of(mockVersion)).when(mockStore).getVersion(1);
+    doReturn(true).when(mockVersion).isActiveActiveReplicationEnabled();
+    VeniceServerConfig mockVeniceServerConfig = mock(VeniceServerConfig.class);
+    VeniceProperties mockVeniceProperties = mock(VeniceProperties.class);
+    doReturn(true).when(mockVeniceProperties).isEmpty();
+    doReturn(mockVeniceProperties).when(mockVeniceServerConfig).getKafkaConsumerConfigsForLocalConsumption();
+    doReturn(Object2IntMaps.emptyMap()).when(mockVeniceServerConfig).getKafkaClusterUrlToIdMap();
+    doReturn(Int2ObjectMaps.emptyMap()).when(mockVeniceServerConfig).getKafkaClusterIdToUrlMap();
+    doReturn(1000L).when(mockVeniceServerConfig).getIngestionHeartbeatIntervalMs();
+    OffsetRecord offsetRecord = mock(OffsetRecord.class);
+    PartitionConsumptionState pcs0 = mock(PartitionConsumptionState.class);
+    doReturn(offsetRecord).when(pcs0).getOffsetRecord();
+    doReturn(LEADER).when(pcs0).getLeaderFollowerState();
+    PartitionConsumptionState pcs1 = mock(PartitionConsumptionState.class);
+    doReturn(offsetRecord).when(pcs1).getOffsetRecord();
+    doReturn(LEADER).when(pcs1).getLeaderFollowerState();
+    doReturn(1).when(pcs1).getUserPartition();
+    PubSubTopic pubsubTopic = mock(PubSubTopic.class);
+    doReturn(pubsubTopic).when(offsetRecord).getLeaderTopic(any());
+    doReturn(true).when(pubsubTopic).isRealTime();
+
+    VeniceWriter veniceWriter = mock(VeniceWriter.class);
+    VeniceWriterFactory veniceWriterFactory = mock(VeniceWriterFactory.class);
+    doReturn(veniceWriter).when(veniceWriterFactory).createVeniceWriter(any());
+
+    StoreIngestionTaskFactory ingestionTaskFactory = TestUtils.getStoreIngestionTaskBuilder(storeName)
+        .setStorageMetadataService(mockStorageMetadataService)
+        .setMetadataRepository(mockReadOnlyStoreRepository)
+        .setTopicManagerRepository(mockTopicManagerRepository)
+        .setServerConfig(mockVeniceServerConfig)
+        .setPubSubTopicRepository(pubSubTopicRepository)
+        .setVeniceWriterFactory(veniceWriterFactory)
+        .build();
+    LeaderFollowerStoreIngestionTask ingestionTask =
+        (LeaderFollowerStoreIngestionTask) ingestionTaskFactory.getNewIngestionTask(
+            mockStore,
+            mockVersion,
+            mockKafkaConsumerProperties,
+            () -> true,
+            mockVeniceStoreVersionConfig,
+            0,
+            false,
+            Optional.empty(),
+            null);
+    ingestionTask.setPartitionConsumptionState(0, pcs0);
+    ingestionTask.setPartitionConsumptionState(1, pcs1);
+    Logger mockLogger = mock(Logger.class);
```

Comment:
I'm not a big fan of having tests for logging... I understand in this case the only change is to logging and so the principle of "there should always be a unit test that fails without the change" points towards adding a test for logging. But I feel like this could be treated as an exception... if the modified code is covered by tests, it could be sufficient, no need to actually have a test that fails when the logging is different.

These are just my two cents, and I don't think we need to delete this new test, but this is more of a forward looking comment (I don't think we need to write logging tests in the future...). Feel free to disagree, of course, though.

Reply:
Thanks @FelixGV, read more about this and it makes sense to not have tests on logging and gave it an attempt to follow it from this PR itself. Modified the `maybeSendIngestionHeartbeat` API to be able to test these cases. Let me know what you think.